modules/4_dimensionality_reduction/rna/dimensionality_reduction.qmd

```diff
@@ -476,12 +478,30 @@ if (length(integration_methods) > 0 & num_datasets > 1) {
 # Join layers if not SCT (SCT does not use layers)
 # Dataset layers are automatically dropped unless there is only one dataset (in which case we need to do it)
 if (!grepl(pattern="_SCT", x=default_assay)) {
-  layer_counts = SeuratObject::Layers(sc, search="counts")
-  layer_data = SeuratObject::Layers(sc, search="data")
+  # Existing layers
+  layer_counts = SeuratObject::Layers(sc, assay=default_assay, search="counts")
```

Comment:
Double-check `default_assay` versus the default set and used below in the sketching part.

Reply:
I double-checked and added more explanations as well as better variable names to document the differences between full and sketch dataset.